crypto-pgpainless/src/main/kotlin/app/passwordstore/crypto/PGPKeyManager.kt

```diff
@@ -40,12 +40,17 @@ constructor(
     withContext(dispatcher) {
       runSuspendCatching {
         if (!keyDirExists()) throw KeyDirectoryUnavailableException
-        val incomingKeyRing = tryParseKeyring(key) ?: throw InvalidKeyException
+        val incomingKeyRing = tryParseKeyring(key)
+
+        if (incomingKeyRing is PGPPublicKeyRing) {
+          throw NoSecretKeyException(tryGetId(key)?.toString() ?: "Failed to retrieve key ID")
+        }
```

Comment:
This method intentionally did not forbid public keys because there are use cases where folks encrypt their store to an additional public key to grant read-only access to the store for the owner of that key without having to disclose their own.

Reply:
Thanks, this should be a TODO then.